calcom/linear-to-github
# Add UI to generate tokens, persist label IDs, and deploy webhooks

# Summary

- Add Next.js + Tailwind boilerplate
- Add Linear auth, team picker, and webhook deploy button
   - Persist Linear label/state IDs in DB
- Add link to pre-populated GH token maker (since GH auth does not support wildcard callbacks)
- Add GH repo picker and webhook deploy button
   - Persist repo and webhook secret in DB
- Extend Prisma schema and build scripts to seed/migrate DB tables at deployment

<img width="1176" alt="image" src="https://user-images.githubusercontent.com/36117635/190484581-d936949b-5d07-484d-8b81-3cfdfcdc3a9f.png">

# Test plan

- Deploy [this Railway template](https://railway.app/new/template/L__0PR). Frontend build should fail for now.
- Clone the generated repo. Copy PostgreSQL connection URL. Paste it as `DATABASE_URL` in **.env**.
- Make a temp Linear app (settings > API > Your applications) or ask @tedspare for the `NEXT_PUBLIC_LINEAR_SECRET`.
- Run `pnpm i && pnpm dev`. Visit `localhost:3000`. The UI should load with a clean console.
- Click authorize Linear. Linear consent tab should open. Click Authorize. Should redirect back to app. Access token and team picker should show.
- Click "deploy webhook". Webhook should appear in Linear settings > API > Webhooks.
- Click "generate GH token". Token creation tab should open with scopes pre-populated. Paste token in app. Repo picker should show.
- Choose repo. Deploy webhook button should show.
- Click "deploy webhook". Webhook should appear in GH settings > Webhooks.
